### PR TITLE
Add alternate code path for Get-cpp-PackageInfoFromRepo

### PR DESCRIPTION
**Pull request by chidozieononiwu**
Add alternative code paths depending on parameters which allows for coming changes to common tools.
This change is solely intended to prevent breaking the build while making changes to common across various repos.

### CODE REVIEW

File: eng/scripts/Language-Settings.ps1 [hunk 1]
```diff
@@ -12,12 +12,23 @@ function Get-cpp-PackageInfoFromRepo($pkgPath, $serviceDirectory, $pkgName)
   # Test if the package path ends with the package name (e.g. sdk/storage/azure-storage-common)
   # This function runs in a loop where $pkgPath might be the path to the package and must return 
   # $null in cases where $pkgPath is not the path to the package specified by $pkgName
-  if ($pkgName -ne (Split-Path -Leaf $pkgPath)) { 
+  if ($pkgName -and ($pkgName -ne (Split-Path -Leaf $pkgPath))) { 
     return $null
   }
 
+  if (!$pkgName)
```
Comment (weshaggard) [inline eng/scripts/Language-Settings.ps1:19]: Should we just ignore the passed pkgName in all cases now?

File: eng/scripts/Get-PkgVersion.ps1 [hunk 2]
```diff
@@ -7,26 +7,36 @@ param (
     [string] $PackageName
 )
 
-. ${PSScriptRoot}/SdkVersion-Common.ps1
+$repoRoot = Resolve-Path "$PSScriptRoot/../..";
+. ${repoRoot}\eng\common\scripts\logging.ps1
```
Comment (weshaggard) [inline eng/scripts/Get-PkgVersion.ps1:11]: Lets make sure these paths correct work as expected on non-windows. If they don't we might need to reverse the slashes or maybe switch to Join-Path.

File: eng/scripts/Get-PkgVersion.ps1 [hunk 3]
```diff
@@ -7,26 +7,36 @@ param (
...
+. ${repoRoot}\eng\scripts\SdkVersion-Common.ps1
 
 $versionFileLocation = Get-VersionHppLocaiton `
     -ServiceDirectory $ServiceDirectory `
     -PackageName $PackageName
 
 if (!$versionFileLocation) {
-    $fallback = Get-Content $RepoRoot/sdk/$ServiceDirectory/$PackageName/version.txt
+    $fallbackpath = "$RepoRoot\sdk\$ServiceDirectory\$PackageName\version.txt"
```
Comment (weshaggard) [inline eng/scripts/Get-PkgVersion.ps1:19]: Why wasn't this an issue before? Are we now getting directories that don't necessarily have a version now?

File: eng/scripts/Get-PkgVersion.ps1 [hunk 4]
```diff
@@ -7,26 +7,36 @@ param (
     [string] $PackageName
 )
 
-. ${PSScriptRoot}/SdkVersion-Common.ps1
+$repoRoot = Resolve-Path "$PSScriptRoot/../..";
+. ${repoRoot}\eng\common\scripts\logging.ps1
+. ${repoRoot}\eng\scripts\SdkVersion-Common.ps1
 
 $versionFileLocation = Get-VersionHppLocaiton `
```
Comment (weshaggard) [inline eng/scripts/Get-PkgVersion.ps1:14]: @danieljurek it looks like we have some room for some clean-up in the versioning scripts we use in cpp. We probably should move most of the logic here and in update into common functions in the shared common script. We should have it import the common.ps1 like the other languages. Lastly we probably should fix the typo in this function `Get-VersionHppLocation` :)

Comment (danieljurek) [reply eng/scripts/Get-PkgVersion.ps1:14]: Agreed on moving the common code between Get and Update package version scripts into the larger ecosystem of scripts. 

`Hpp` is not a typo. It's the extension we're using for C++ header files which contains the explicit version constants ([example](https://github.com/Azure/azure-sdk-for-cpp/blob/master/sdk/core/azure-core/inc/azure/core/version.hpp))

Comment (danieljurek) [reply eng/scripts/Get-PkgVersion.ps1:14]: https://github.com/Azure/azure-sdk-for-cpp/issues/1764

File: eng/scripts/Get-PkgVersion.ps1 (hunk 3 above)
Comment (danieljurek) [reply eng/scripts/Get-PkgVersion.ps1:19]: In times past we used a `version.txt` file before the language standard called for the version information be stored in `version.hpp` ... There was a time when some of the packages used `version.txt`. However, this code path should not execute anymore as all packages have migrated off of `version.txt`.

I'm tracking removal in: https://github.com/Azure/azure-sdk-for-cpp/issues/1764

File: eng/scripts/Get-PkgVersion.ps1 (hunk 4 above)
Comment (weshaggard) [reply eng/scripts/Get-PkgVersion.ps1:14]: Hpp isn't the typo it is the `Locaition` vs `Location`

File: eng/scripts/Get-PkgVersion.ps1 [hunk 5]
```diff
@@ -8,15 +8,15 @@ param (
 )
 
 $repoRoot = Resolve-Path "$PSScriptRoot/../..";
-. ${repoRoot}\eng\common\scripts\logging.ps1
-. ${repoRoot}\eng\scripts\SdkVersion-Common.ps1
+. (Join-Path ${repoRoot} "eng" "common" "scripts" "logging.ps1")
```
Comment (weshaggard) [inline eng/scripts/Get-PkgVersion.ps1:11]: nit: all the quotes aren't needed.

**Review by weshaggard — APPROVED** [review]

**Pull request merged**